tests/parametric/test_dynamic_configuration.py

```diff
@@ -796,3 +808,62 @@ def test_trace_sampling_rules_with_tags(self, library_env, test_agent, test_libr
         span = trace[0]
         assert "_dd.p.dm" in span["meta"]
         assert span["meta"]["_dd.p.dm"] == "-12"
+
+    @parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_remote_sampling_rules_telemetry(self, library_env, test_agent, test_library):
+        """RC sampling rules should be reported by telemetry"""
+        event = set_and_wait_rc_telemetry(
+            test_agent,
+            config_overrides={
+                "tracing_sampling_rules": [
+                    {
+                        "service": "svc*",
+                        "resource": "*abc",
+                        "name": "op-??",
+                        "tags": [
+                            {"key": "tag-a", "value_glob": "ta-v*"},
+                            {"key": "tag-b", "value_glob": "tb-v?"},
+                            {"key": "tag-c", "value_glob": "tc-v"},
+                        ],
+                        "sample_rate": 0.5,
+                        "provenance": "dynamic",
+                    }
+                ],
+            },
+        )
+        configuration = event["payload"]["configuration"]
+        assert configuration == str(
+            [{"service": "svc*", "resource": "*abc", "name": "op-??", "sample_rate": 0.5, "provenance": "dynamic"}]
+        ) or str(
```

Comment:
This test was passing for me even though the strings did not match - I realized that this is actually checking
`assert configuration == str(...)` **OR** `str(...)` - and the truthiness of the second string literal was evaluating to `true` - I'd recommend adding parentheses around the whole condition being asserted

Also the configuration from telemetry will contain the `"tracing_sampling_rules":` key - so maybe we just need to check the returned configuration contains the expected sequence?

Reply:
see alternative suggestion below which uses `json.loads`